player.py

```diff
@@ -32,19 +32,26 @@ def make_move(self, state):
 
         pawn = None
         valid = False
-        print('Begin by selecting a pawn in a row and column')
-        while not valid:
-            row, col = eval(input('row, col: '))
-            pawn = state.get_pawn(row, col, self.color)
-            if pawn is not None:
-                valid = True
         final_pos = None
-        valid = False
-        print('Select a final position for the pawn')
         while not valid:
-            row, col = eval(input('row, col: '))
-            if state.check_cell(row, col) == label.BLANK:
-                final_pos = Position(row, col)
-                valid = True
+            print('Begin by selecting a pawn in a row and column')
+            row = eval(input('Select a row: '))
+            col = eval(input('Select a column: '))
+            if (row in range(3)) and (col in range(3)):
+                pawn = state.get_pawn(row, col, self.color)
+                if pawn is not None:
+                    valid = True
```

Comment:
Although I enjoy the idea of merging these two input loops, I think require an extra step from the user if the row and column selections are invalid. Consider when the `row` and `col` do not refer to a valid pawn. Then, the user would proceed to provide a final position even though the original pawn position is invalid!

Can we avoid this scenario?

Reply:
It is true that the first loop would allow the user to continue to the second loop without a valid original position, however the second loop checks **both** the original position and the proposed final position using your 'get_available_moves' function. Therefore, even if the player doesn't select a valid original position, it will still be rejected at the second loop, only moves out of the legal moveset list will be allowed. I do agree that this does make the first loop kind of redundant as a 'check' however I left it in operation in order to grab the 'pawn' variable.

Reply:
I am okay with this resolution, but I do see one place we can simplify the code in lines 42 to 43: the conditional statement can be merged into one line.

```python
valid = pawn is not None
```

Reply:
Yes this is a good simplification.
Also I propose; like I suggested earlier the first loop is redundant as a check so it would be a good idea to wrap the second loop in an 'if' so we can actually use this loop as a check. 
`if valid == True:`
This way, if the user does not select a valid position, it will not bother asking for a final position it will skip this and ask the user to try again as soon as it knows the move will not work. 
If you think this is a good idea I will push the commit, if you think it is unnecessary and adds too much complication then I think it's best to remove the validation from the first loop since it will be checked in the second loop regardless.

Reply:
I like your idea. :)